From 09c954bff3a3d835f00894b0ccb12507a34b5287 Mon Sep 17 00:00:00 2001
From: Koen Zandberg <koen@bergzand.net>
Date: Wed, 20 Jun 2018 16:53:45 +0200
Subject: [PATCH] Use random_bytes instead

---
 tweetnacl.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tweetnacl.c b/tweetnacl.c
index 360bcfc8f..e6e58c67e 100644
--- a/tweetnacl.c
+++ b/tweetnacl.c
@@ -1,4 +1,5 @@
 #include "tweetnacl.h"
+#include "random.h"
 #define FOR(i,n) for (i = 0;i < n;++i)
 #define sv static void
 
@@ -7,7 +8,6 @@ typedef unsigned long u32;
 typedef unsigned long long u64;
 typedef long long i64;
 typedef i64 gf[16];
-extern void randombytes(u8 *,u64);
 
 static const u8 _0[16];
 
@@ -455,7 +455,7 @@ int crypto_scalarmult_base(u8 *q,const u8 *n)
 
 int crypto_box_keypair(u8 *y,u8 *x)
 {
-  randombytes(x,32);
+  random_bytes(x,32);
   return crypto_scalarmult_base(y,x);
 }
 
@@ -665,7 +665,7 @@ int crypto_sign_keypair(u8 *pk, u8 *sk)
   gf p[4];
   int i;
 
-  randombytes(sk, 32);
+  random_bytes(sk, 32);
   crypto_hash(d, sk, 32);
   d[0] &= 248;
   d[31] &= 127;
-- 
2.16.4

